raft/src/test/java/org/apache/kafka/raft/RaftEventSimulationTest.java

```diff
@@ -406,7 +446,10 @@ private SequentialAppendAction(Cluster cluster) {
 
         @Override
         public void execute() {
-            cluster.withCurrentLeader(node -> node.counter.increment());
+            cluster.withCurrentLeader(node -> {
+                if (!node.client.isShuttingDown())
```

Comment:
Should we use `isRunning` instead? Could it be possible that the client is already completed shutdown?

Reply:
I think the reason I did this is because we currently reject appends on nodes that are shutting down. Let me think about loosening this.

Reply:
I decided to leave this as is since reworking the `pollShutdown` logic is a bit more work. Let's consider this in the context of some of the state machine refactors.

Reply:
SG.